From 95031a64e3d5b0894a91b299ba47c7b95ed3fcbd Mon Sep 17 00:00:00 2001
From: =?utf-8?q?J=C3=B6rg=20Sommer?= <joerg@alea.gnuu.de>
Date: Thu, 21 Aug 2008 14:50:43 +0200
Subject: [PATCH 5/5] Fetch only last version if version parameter is last

It's much more handy, especially in scripts, to fetch only the last
version without having to know the version number.
---
 getbuildlog   |   17 +++++++++++++++--
 getbuildlog.1 |    4 +++-
 2 files changed, 18 insertions(+), 3 deletions(-)

diff --git a/getbuildlog b/getbuildlog
index d1e7508..a005ad8 100755
--- a/getbuildlog
+++ b/getbuildlog
@@ -28,7 +28,8 @@ Usage: $PROGNAME <package> [<version-pattern>] [<architecture-pattern>]
   Downloads build logs of <package> from Debian auto-builders.
   If <version-pattern> or <architecture-pattern> are given, only build logs
   whose versions and architectures, respectively, matches the given patterns
-  are downloaded.
+  are downloaded. If <version-pattern> is last, only download the build logs
+  of the last version.
 Options:
   -h, --help        Show this help message.
   -V, --version     Show version and copyright information.
@@ -69,6 +70,13 @@ VERSION=${2:-[:~+.[:alnum:]-]+}
 ARCH=${3:-[[:alnum:]-]+}
 ESCAPED_PACKAGE=`echo "$PACKAGE" | sed -e 's/\+/\\\+/g'`
 
+if [ "$VERSION" = last ]; then
+    GET_ONLY_LAST_VERSION=true
+    VERSION='[:~+.[:alnum:]-]+'
+else
+    GET_ONLY_LAST_VERSION=false
+fi
+
 PATTERN="fetch\.(cgi|php)\?&pkg=$ESCAPED_PACKAGE&ver=$VERSION&arch=$ARCH&\
 stamp=[[:digit:]]+"
 
@@ -84,9 +92,14 @@ getbuildlog() {
     # Quick-and-dirty unescaping
     sed -i -e "s/%2B/\+/g" -e "s/%3A/:/g" -e "s/%7E/~/g" $ALL_LOGS
 
-    for match in `grep -E -o "$PATTERN" $ALL_LOGS`; do
+    last_ver=
+    for match in `grep -E -o "$PATTERN" $ALL_LOGS | tac`; do
         ver=${match##*ver=}
         ver=${ver%%&*}
+        if $GET_ONLY_LAST_VERSION && [ "${last_ver:-$ver}" != "$ver" ]; then
+            break
+        fi
+        last_ver=$ver
         arch=${match##*arch=}
         arch=${arch%%&*}
 	match=`echo $match | sed -e 's/\+/%2B/g'`
diff --git a/getbuildlog.1 b/getbuildlog.1
index 9861332..0376600 100644
--- a/getbuildlog.1
+++ b/getbuildlog.1
@@ -11,7 +11,9 @@ auto\-builders. It downloads build logs of all versions and for all
 architectures if \fIversion\-pattern\fR and \fIarchitecture\-pattern\fR are
 not specified or empty, otherwise only build logs whose versions match
 \fIversion-pattern\fR and build logs whose architectures match
-\fIarchitecture-pattern\fR will be downloaded. The version and architecture
+\fIarchitecture-pattern\fR will be downloaded. If \fIversion\-pattern\fP is
+last, only download the build logs of the last version. The version and
+architecture
 patterns are interpreted as extended regular expressions as described in
 \fBgrep\fR(1).
 .SH OPTIONS
-- 
1.5.6.3

